modules/cli/src/main/scala/scala/cli/commands/util/SharedOptionsUtil.scala

```diff
@@ -156,10 +156,16 @@ object SharedOptionsUtil extends CommandHelpers {
       jmhVersion: Option[String] = None,
       ignoreErrors: Boolean = false
     ): bo.BuildOptions = {
-      val platformOpt =
-        if (js.js) Some(Platform.JS)
-        else if (native.native) Some(Platform.Native)
-        else None
+      val platformStr = platform.map(Platform.normalize)
+      val platformOpt = platformStr.map(Platform.parse) match {
+        case Some(p) => p
+        case None =>
+          if (native.native && js.js) Some(Platform.Native)
```

Comment:
Instead of the complex `if (..) else if (..) else` clause, you could solve the whole thing with asingle `match` clause.

So something like:
```scala
val platformOpt = (platformStr.map(Platform.parse), native.native, js.js) match {
        case (Some(p), _, _) => p
        case (_, true, _) => Some(Platform.Native)
        case (_, _, true) => Some(Platform.JS)
        case _ => None
      }
```

Reply:
also, I'm thinking if we shouldn't throw an error if conflicting platforms are picked, instead of guessing that one of them should have priority.

Reply:
I tried to raise a warning saying about platform ambiguity, but buildOptions function is triggered 3 times and the warning was raised every single time. I didn't have idea how to do it properly and that is why I gave up with it.

Reply:
But error is quite a good idea, since it will finish scala-cli on the first call of buildOptions.

Reply:
yeah, I don't think a warning is sufficient here. 
IMO specifying conflicting platforms should not be treated as a valid command and fail instead.